maven/maven.bzl

```diff
@@ -229,19 +231,44 @@ _generate_maven_repository = repository_rule(
         "maven_rules_repository": attr.string(mandatory = False, default = "maven_repository_rules"),
         "dependency_target_substitutes": attr.string_list_dict(mandatory = True),
         "build_snippets": attr.string_dict(mandatory = True),
+        "use_jetifier": attr.bool(default = False),
     },
 )
 
 # Implementation of the maven_jvm_artifact rule.
-def _maven_jvm_artifact(artifact_spec, name, visibility, deps = [], **kwargs):
+def _maven_jvm_artifact(artifact_spec, name, visibility, deps = [], use_jetifier = False, **kwargs):
     artifact = artifacts.annotate(artifacts.parse_spec(artifact_spec))
     maven_target = "@%s//%s:%s" % (artifact.maven_target_name, _DOWNLOAD_PREFIX, artifact.path)
     import_target = artifact.maven_target_name + "_import"
     target_name = name if name else artifact.third_party_target_name
+    coordinate = "%s:%s" % (artifact.group_id, artifact.artifact_id)
+
+    should_jetify = use_jetifier and coordinate not in [
+      "javax.annotation:jsr250-api",
+      "com.google.code.findbugs:jsr305",
+      "com.google.errorprone:javac-shaded",
+      "com.google.googlejavaformat:google-java-format",
+      "com.squareup:javapoet",
+      "com.google.dagger:dagger-compiler",
+      "com.google.dagger:dagger-producers",
+      "com.google.dagger:dagger-spi",
+      "com.google.dagger:dagger",
+      "javax.inject:javax.inject",
+    ] and artifact.group_id != "org.bouncycastle" and not artifact.group_id.startswith("androidx")
```

Comment:
Can you extract this coordinate list into an EXCLUDED_ARTIFACTS constant please? 

You might be able to make a "EXCLUDED_GROUP_PREFIXES" too and do the later conditional logic in a more holistic way, but leave it for now.  Two conditions doesn't make it worthwhile. If we come up with a larger set of excluded groups, we can refactor it then.

Reply:
Done and done and discussed.